File: 3_16/3_16.py
def check_date(num):
    days_in_month = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
    day, month = 0, 0
    if num >= 11 and num <= 99:
        return True
    if num >= 101 and num <= 999:
        day = num // 10
        month = num % 10
        if days_in_month[month - 1] >= day:
            return True
        day = num // 100
        month = num % 100
        if days_in_month[month - 1] >= day:
            return True
    if num >= 1000 and num < 9999:
        day = num // 100
        month = num % 100
        if days_in_month[month - 1] >= day:
            return True
    return False

count = int(input())
numbers = []
for i in range(count):
    n = int(input())
    numbers.append(n)

for n in numbers:
    if check_date(n):
        print("Yes")
    else:
        print("No")

File: 3_16/3_16.rs
use std::io;

// Function to check if the given number can represent a valid date
fn check_date(num: i32) -> bool {
    let mut i = 1;
    while i < num {
        let day = num / i;
        let month = num % i;
        if day >= 1 && day <= 31 && month >= 1 && month <= 12 {
            return true;
        }
        i *= 10;
    }
    false
}

fn main() {
    let mut input_string = String::new();
    io::stdin().read_line(&mut input_string).expect("Failed to read line");
    let count: usize = input_string.trim().parse().expect("Input not an integer");

    let mut numbers = Vec::new();
    for _ in 0..count {
        input_string.clear(); // Clear the string to reuse it for the next number
        io::stdin().read_line(&mut input_string).expect("Failed to read line");
        let num: i32 = input_string.trim().parse().expect("Input not an integer");
        numbers.push(num);
    }

    for &num in &numbers {
        if check_date(num) {
            print!("YES ");
        } else {
            print!("NO ");
        }
    }
}